### PR TITLE
Add Course Edit/Create forms

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**

### CODE REVIEW

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]
remove package-lock.json and use yarn

File: pages/admin/courses/index.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  async asyncData({ $axios }) {
+    const {
+      data: { data: courses },
+    } = await $axios.get('/courses')
+    return { courses }
+  },
+  methods: {
+    addNewCourse() {
+      const newCourse = {
```
Comment (jatinkatyal13) [inline pages/admin/courses/index.vue]: Remove

File: pages/admin/courses/add.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    async addNewCourse() {
+      // api call to backend POST
+      try {
+        await this.$axios.$post(
+          '/admin/courses',
+          {
+            ...this.model,
+          }
+        )
+        alert('Saved Successfully')
```
Comment (jatinkatyal13) [inline pages/admin/courses/add.vue]: redirect to new page after save

File: pages/admin/courses/add.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
+<template>
+  <div class="container">
+    <form>
+      <vue-form-generator
+        :schema="schema"
+        :model="model"
+        :options="formOptions"
+        @validated="onNewCourse"
```
Comment (jatinkatyal13) [inline pages/admin/courses/add.vue]: `onNewCourse` isn't it undefined?

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]

File: pages/admin/courses/_id.vue [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                inputType: 'text',
+                label: 'Slug',
+                model: 'slug',
+                help: `This will be the <a href="https://stackoverflow.com/questions/19335215/what-is-a-slug"> slug </a> for the course.
+                  Use kebab-case (hyphen separated) name and avoid special characters if you can) For example: c++-for-beginners`,
+                placeholder: 'This will be the slug for the course',
+                required: true,
+              },
+              {
+                type: 'textArea',
```
Comment (jatinkatyal13) [inline pages/admin/courses/_id.vue]: course syllabus is a JSON object and will have a complex editor, we can either skip this as of now or create a component that can edit different fields in the syllabus starting with just the title

File: components/AdminNavbar.vue [hunk 5]
```diff
@@ -1,31 +1,73 @@
...
 export default Vue.extend({
   name: 'AdminNavbar',
   components: {
-    Button
-  }
+    Button,
+  },
+  methods: {
+    openNavbar() {
+      const navbar = document.getElementById('navbar')
```
Comment (abhishek97) [inline components/AdminNavbar.vue:76]: Can you explain more what's the intention here? As a rule of thumb you shouldn't be using DOM API directly unless you're dealing with some third party lib that doesn't understand vue.

Comment (aggarwalpulkit596) [reply components/AdminNavbar.vue:76]: @jatinkatyal13

File: pages/admin/courses/_id.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        :options="formOptions"
+      >
+      </vue-form-generator>
+    </form>
+  </div>
+</template>
+ 
+<script>
+import Vue from 'vue'
+const VueFormGenerator = require('vue-form-generator')
```
Comment (abhishek97) [inline pages/admin/courses/_id.vue]: why not `import` ?  Since this is built via webpack, we can just stick with es6 imports for consistency

Comment (aggarwalpulkit596) [reply pages/admin/courses/_id.vue]: For some reason it wasn't working I couldn't figure out why that's why i used this.

File: pages/admin/courses/_id.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+import 'vue-form-generator/dist/vfg.css'
+
+Vue.use(VueFormGenerator)
```
Comment (abhishek97) [inline pages/admin/courses/_id.vue]: This code belong in a plugin. See https://github.com/coding-blocks/amoeba-frontpages/blob/master/online.codingblocks.com/plugins/vuency.js

File: components/AdminNavbar.vue (hunk 5 above)
Comment (jatinkatyal13) [reply components/AdminNavbar.vue:76]: I am using our side menu web component here, and hence need this to show/hide it. Maybe it'll be removed in the future or we'll just setup a new web component for the hamburger menu

File: pages/admin/courses/add.vue [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+      },
+    }
+  },
+  methods: {
+    async addNewCourse() {
+      // api call to backend POST
+      try {
+        await this.$axios.$post(
+          '/admin/courses',
+          {
```
Comment (abhishek97) [inline pages/admin/courses/add.vue]: why instantiate a new Object here?

**Pull request merged**